standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java

```diff
@@ -6993,11 +6993,16 @@ public GetPartitionsByNamesResult get_partitions_by_names_req(GetPartitionsByNam
           // request them separately.
           for (Partition part: ret) {
             String partName = Warehouse.makePartName(table.getPartitionKeys(), part.getValues());
-            List<ColumnStatistics> partColStatsList =
-                    getMS().getPartitionColumnStatistics(parsedCatName, parsedDbName, tblName,
-                            Collections.singletonList(partName),
-                            StatsSetupConst.getColumnsHavingStats(part.getParameters()),
-                            engine);
+            List<ColumnStatistics> partColStatsList = null;
+            List<String> colNames = StatsSetupConst.getColumnsHavingStats(part.getParameters());
```

Comment:
Might be a good idea to push this check inside the getPartitionColumnStatistics

Reply:
Agree, I will make the change